docs/source/reference/tpu.rst

```diff
@@ -0,0 +1,188 @@
+.. _tpu:
+
+Cloud TPU
+================================
+
+SkyPilot supports running jobs on Google's `Cloud TPU <https://cloud.google.com/tpu/docs/intro-to-tpu>`_.
+Two different TPU architectures are available on GCP:
+
+- `TPU Nodes <https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#tpu-node>`_
+- `TPU VMs <https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#tpu-vm>`_
+
+Both are supported by SkyPilot. The two architectures differ as follows.
+For TPU Nodes, a host VM communicates with the TPU host over gRPC.
+For TPU VMs, you can SSH directly into a VM that is physically connected to the TPU device.
+For more details please refer to GCP `documentation <https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#tpu-arch>`_.
+
+
+.. note::
+
+   We encourage researchers to apply free TPU access through `TPU Research Cloud (TRC) <https://sites.research.google/trc/about/>`_ program.
+
+
+Getting TPUs in one command
+--------------------------------
+
+Like :ref:`GPUs <interactive-nodes>`, SkyPilot provides a simple command to quickly get TPUs for development:
+
+.. code-block:: bash
+
+   sky tpunode                                # By default TPU v3-8 is used
+   sky tpunode --use-spot                     # Preemptible TPUs
+   sky tpunode --tpus tpu-v2-8                # Change TPU type to tpu-v2-8
+   sky tpunode --instance-type n1-highmem-16  # Change the host VM type to n1-highmem-16
+   sky tpunode --tpu-vm                       # Use TPU VM (instead of TPU Node)
+
+After the command has finished, you will be dropped into the host VM and can start develop code right away!
+
+Below we demonstrate how to run MNIST training on both TPU Nodes and TPU VMs with SkyPilot YAML.
+
+TPU Nodes
+--------------------------------
+
+To use TPU Node, a host CPU VM needs to be created together with a TPU node and configured correctly to connect with each other.
+SkyPilot automates the above process with a simple interface:
+
+.. code-block:: yaml
+
+   resources:
+      instance_type: n1-highmem-8
+      accelerators: tpu-v2-8
```

Comment:
Just curious: why are we using `tpu-v2-8` while our default TPU device is `tpu-v3-8`? Is there a particular reason for this?

Reply:
Great point. I thought about it and there are three reasons we might want to change our default tpunode to `v2-8`.
1. it's considerably harder to get `v3-8` than `v2-8`.  I've met many out-of-capacity issues (in all regions) when requesting a `v3-8` (occurred to me last night and this morning). It can be a sad first impression if that happens to first-time users.
2. `v2-8` is $4.5/hr vs `v3-8` $8/hr. for development/demo purpose we want to use cheaper resources
3. No code compatibility issue. Almost all codes work on both `v2-8` and `v3-8`. except the latter has larger memory.

WDYT? @WoosukKwon @concretevitamin

Reply:
Sgtm. (v2-8 may be hard to get as well, however; we can talk offline.)

Reply:
@infwinston If that's the case, I think changing the default tpu to `v2-8` would be fine. Either choice works for TRC users as TRC offers 5 TPU-v3 devices and 5 TPU-v2 devices.

Reply:
Sounds good I just updated it.